.github/workflows/auto-merge.yml

```diff
@@ -0,0 +1,36 @@
+name: Merge stable into child branches
+
+on:
+  push:
+    branches: ["stable"]
+
+jobs:
+  create-pull-request:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        branch: [minor-next, major-next]
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # Fetch all branches and their history
+
+      - name: Fetch all branches
+        run: git fetch --all
+
+      - name: Merge changes
+        run: |
+          git checkout ${{ matrix.branch }}
+          git merge stable
```

Comment:
The usual merge strategy is

`stable` -> `minor-next`
`minor-next` -> `major-next`

It doesn't make sense to use a matrix for this, since the branches should be merged sequentially.

This also doesn't make any considerations for if a conflict arises.

Reply:
If a conflict arises, this will just fail the CI. Do we need to do anything about it?

Reply:
It still shouldn't be using a matrix as the branches need to be sequentially merged.

This workflow will also generate 2 PRs for every commit pushed, which seems to me like it would be a major annoyance.